src/TestFramework/TestFramework.Extensions.Csv/CsvDataSourceAttribute.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Data;
+using System.Data.OleDb;
+using System.Globalization;
+using System.Reflection;
+
+namespace Microsoft.VisualStudio.TestTools.UnitTesting;
+
+/// <summary>
+/// Attribute to define dynamic data from a CSV file for a test method.
+/// </summary>
+[AttributeUsage(AttributeTargets.Method, AllowMultiple = false/*TODO: Do we want to allow multiple*/)]
```

Comment:
I think we allow multiple `DynamicData` so we should probably do the same.

Reply:
Agreed. Done